eng/pipeline/stages/run-stage.yml

```diff
@@ -75,13 +75,9 @@ stages:
                 container: mariner2
               ${{ elseif eq(parameters.builder.distro, 'azurelinux3') }}:
                 container: azurelinux3
-              ${{ else }}:
-                container: mariner1
```

Comment:
Removing the `else` condition means that the jobs that would have hit this condition will run with no container. I believe the host machines are a recent Ubuntu. (This might be why `linux-amd64 test [opensslcrypto] FIPS` is failing.)

I think we should actually make mariner 2 the default, and we don't need `mariner2`-specific builders anymore.

https://github.com/microsoft/go/blob/f10f06228f690238ddbf07515edd06c1e8e0d20f/eng/pipeline/stages/go-builder-matrix-stages.yml is important here because it feeds into these conditions.